modules/util/source-formatter/src/main/java/com/liferay/source/formatter/checkstyle/check/DatabaseMetaDataCheck.java

```diff
@@ -23,6 +23,14 @@ public int[] getDefaultTokens() {
 
 	@Override
 	protected void doVisitToken(DetailAST detailAST) {
+		String absolutePath = getAbsolutePath();
+
+		if (absolutePath.contains("/com/liferay/portal/dao/db/") &&
+			absolutePath.endsWith("DB.java")) {
+
```

Comment:
The SF rule does what we need but we need to be more accurate regarding when we can skip the check. Can you do it only when the class fulfills this condition but also when the getIndexInfo call is only in a method called getIndexResultSet?

Thanks

Reply:
@achaparro, that should be simple to do. I will add that condition.

Reply:
@achaparro, just to make sure I understand correctly, we want to check if `getIndexInfo()` is in a `getIndexResultSet()` method in a DB class? Like this?

```java
String absolutePath = getAbsolutePath();

if (absolutePath.contains("/com/liferay/portal/dao/db/") &&
	absolutePath.endsWith("DB.java")) {

	DetailAST methodDefinitionDetailAST = getParentWithTokenType(
		detailAST, TokenTypes.METHOD_DEF);

	if ((methodDefinitionDetailAST != null) &&
		Objects.equals(
			getName(methodDefinitionDetailAST), "getIndexResultSet")) {

		return;
	}
}
```

Or do we check for `getIndexInfo()` in __all__ cases?

Reply:
@kevhlee, yes, only in *DB classes. The only place where the call to getIndexInfo is valid is in getIndexResultSet

I am not an expert in SF rule but it looks good :-)

Thank you!

Reply:
@achaparro, that makes sense. I will make the change and resend.